api/v1alpha1/envoygateway_types.go

```diff
@@ -558,6 +558,12 @@ type ExtensionService struct {
 	//
 	// +optional
 	TLS *ExtensionTLS `json:"tls,omitempty"`
+
+	// RetryPolicy defines the retry policy for to use when errors are encountered in communication with
+	// the extension service.
+	//
+	// +optional
+	RetryPolicy *RetryPolicy `json:"retryPolicy,omitempty"`
```

Comment:
can we reuse struct from gwapi [gep-1731](https://github.com/kubernetes-sigs/gateway-api/pull/3199)?

Reply:
The scope of the GW-API retry policy seems limited HTTPRoutes, and gRPC status codes are intentionally left out:

https://gateway-api.sigs.k8s.io/geps/gep-1731/

> Non-goals:
> ...
> * To allow specification of gRPC status codes for which a request should be retried (this should be covered in a separate GEP).

Once supported, we can deprecate our custom struct.